src/WebLinks/AzureDevOpsWebProvider.cs

```diff
@@ -70,5 +60,37 @@ public string CreateFileUrl(string repositoryUrl, string relativePath, VersionIn
 
             return sb.ToString();
         }
+
+        private bool IsSshUri(string repositoryUrl)
+        {
+            Match match = Regex.Match(repositoryUrl, SshUriPattern);
+            return match.Success;
+        }
+
+        private bool IsHttpUri(string repositoryUrl)
+        {
+            var uri = new Uri(repositoryUrl);
+            string hostName = uri.Host;
+
+            return StringComparer.OrdinalIgnoreCase.Equals(hostName, "dev.azure.com")
+                || hostName.EndsWith(".visualstudio.com", StringComparison.OrdinalIgnoreCase);
+        }
+
+        private string ExtractBaseFileUrl(string repositoryUrl)
+        {
+            if(IsSshUri(repositoryUrl))
+            {
+                Match match = Regex.Match(repositoryUrl, SshUriPattern);
+                string organization = match.Groups["organization"].Value;
+                string project = match.Groups["project"].Value;
+                string repo = match.Groups["repo"].Value;
+
+                return $"https://{organization}.visualstudio.com/DefaultCollection/{project}/_git/{repo}";
```

Comment:
Technically the `DefaultCollection` part of vs.com URLs is no longer required, please remove it.

/aside

If you want to you can try and create the URL for the same host name that the remote URL is for (see comment above about the new regex which can match any host name).

In other words, if the user has an SSH remote of `git@dev.azure.com:v3/org/project/repo` then the base file URL would be `https://dev.azure.com/org/project/_git/repo` rather than always `visualstudio.com`.

However I'm happy to studiously look the other way about this part! You can stick with always returning `{org}.visualstudio.com`. 😁

Reply:
good point, let's stick with the `{org}.visualstudio.com` for now and delegate the multi domain support work to Azure :)